grpc/interceptor/interceptor.go

```diff
@@ -87,13 +88,16 @@ func fieldsFromContext(ctx context.Context) log.Fields {
 		}
 	}
 
-	md := api.MetadataFromContext(ctx)
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
```

Comment:
this won't work, if the context is reused in another request, as then both "Incoming" and "Outgoing" contexts will be present, right?
suppose we have 3 services, where context is reused, say `B` adds additional fields to `ctx` it received from `A`.
```
A --ctx--> B --ctx--> C
```
at `B --> C` ctx includes both incoming(from `A`) and outgoing context (from `B`), or am I wrong?